pychromecast/socket_client.py

```diff
@@ -273,7 +286,8 @@ def mdns_backoff(service, retry):
                     continue
                 try:
                     if self.socket is not None:
-                        self.socket.close()
+                        # If we retry connecting, we need to clean up the socket again
+                        self.socket.close()  # type: ignore[unreachable]
```

Comment:
Why is this unreachable?

Reply:
I don't think it is, I think mypy gets confused because we set it to None here: https://github.com/home-assistant-libs/pychromecast/blob/518187629f4980504f1e3aaf1886040fa2f8ef5a/pychromecast/socket_client.py#L241

Reply:
Ok.